clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -791,6 +791,23 @@ public void testNodeIfOffline() {
         assertEquals(metadata.fetch().nodeById(1).id(), 1);
     }
 
+    @Test
+    public void testNodeIfOnlineNonExistentTopicPartition() {
+        Node node0 = new Node(0, "localhost", 9092);
+
+        MetadataResponse metadataResponse = RequestTestUtils.metadataUpdateWith("dummy", 2, Collections.emptyMap(), Collections.emptyMap(), _tp -> 99,
```

Comment:
Could we use `RequestTestUtils.metadataUpdateWith(2, Collections.emptyMap())`? It seems to me that we would end up with the same result because of the empty maps.

Reply:
Thank you for pointing out this test helper, I made the change.